### DISCUSSION TITLE
Feature request: IP reservation

### DISCUSSION THREAD

**Discussion by sebcsaba** [mattlongman/Hassio-Access-Point#41] [Ideas]
I'd ask for a configuration option to add dhcp reservations, simly mac-ip pairs. According to dsnmasq's dhcp-host config feature. This could help reassign the same IP for the devices already set up for the Hass.io

**Comment by ex-ml** [comment]
Hi @sebcsaba - this should be possible as it stands, although without an explicit config option for it.

The addons uses dnsmasq for DHCP, so the dnsmasq override option can be used to reserve IPs.

From https://www.linux.com/topic/networking/dns-and-dhcp-dnsmasq/:

>## Set Static Addresses from DHCP
>
>This is my favorite thing. You may assign static IP addresses to your LAN hosts by MAC address, or by hostname. The address must fall in a range you have already configured with dhcp-range=:
>
>dhcp-host=d0:50:99:82:e7:2b,192.168.10.46
>dhcp-host=turnip,192.168.10.45

You might need to test the exact syntax, but adding `dhcp-host=MACADDRESS,IP-You-Want-To-Reserve` to the dnsmasq override is where I'd start. Note that the IP you want to reserve needs to be within the DHCP range already specified in the config, otherwise dnsmasq can't assign it to your device.

If you (or anyone else) wants to add a specific config option to the addon, this is where to begin.

Hope this helps!
